docs/git.rst

```diff
@@ -0,0 +1,611 @@
+  am I missing something?
+- *others*
+
+If you can not express what has been changed within 50 characters
+(suggested length of a commit message subject),
+or you say it like "it does XXX and YYY",
+you most probably need to split that commit into,
+at least,
+two or more commits.
+
+That doesn't mean that a +20 files or +100 lines of code changes are bad per se,
+you may be doing a simple refactoring across lots of files,
+that's fine and good actually.
+
+As long as a commit is just and only about a specific purpose,
+and not a mixed selection of the following:
+
+- refactoring code
+- moving things around
+- fixing some bugs while at it
+- adding some docs
+- a new cool feature
+- fixing typos on documentation
+- pep8 fixes
+
+Is completely fine to make refactorings.
+
+And this is actually to help both your present self and your +5 years from now that will have to refactor that code of yours,
+and maybe is struggling to understand what was going on there.
+
+Following this advice will:
+
+- keep things simple where there's no gain on adding complexity
+- make your changes easy to be reviewed
+- make later on lookups on those changes easy to follow
+
+Making commits
+--------------
+For commit messages see:
+`plone API guidelines <http://docs.plone.org/external/plone.api/docs/contribute/conventions.html#git-commit-message-style>`_.
+
+Adding references to issues
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Always add the full URL to the issue/pull request you are fixing/referring to.
+
+Maybe within the git repository it makes sense,
+but as soon as you are outside of it,
+it will not.
+
+Take into account mr.roboto automatic commits to buildout.coredev for example,
+if your commit message goes like *Fix for #33*,
+which issue/pull request is that fixing?
+The one in buildout.coredev itself?
+On another issue tracker?
+Somewhere else?
+
+It would be far better if the commit goes instead like::
+
+    Brief description
+
+    Further explanation.
+
+    Fixes: https://github.com/plone/plone.app.discussion/issue/999
+
+Bad examples
+^^^^^^^^^^^^
+Some bad examples of commit messages:
+https://github.com/plone/plone.app.content/commit/0f3a6c65b2018e0ecc65d0ad1581e345f17e531b
+
+Commit messages goes like *"Make note about how this interface is now for BBB only"*.
+
+Question:
+if it's BBB only,
+where is the new place to look for that interface now?
+
+The problem is that,
+in this case Martin,
+wrote that in 2009,
+so most probably once a refactor of that package is done later on 2015,
+Martin is no longer around,
+and if he would be,
+most probably he would not remember something from +6 years ago.
+
+Ask yourself a question:
+if someone comes to you asking details about a random commit done by you +5 years ago,
+what will you reply?
+
+Try that,
+get one project that you worked 5 years ago,
+get a random commit and:
+
+- see if by just reading the commit message already gives enough information of what changes are in
```

Comment:
"See if, just by reading the commit message, you are given enough information of what changes have been made, when comparing the commit message and the actual code. Does the commit message match the code changed?"